### PR TITLE
chore: revamp enable screen

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Redesign of "Enable" screen

### Type of change (Remove other not matching type)

- `chore`: Revamp (non-breaking change improving design)

## 1) Enable Screen

<p>

</p>

## 2) lnurlAuth Screen

<p>

</p>

## 3) lnurlPay Screen

<p>

</p>

## 4) lnurlWithdraw Screen

<p>

</p>

## 5) MakeInvoice Screen

Screenshot from 2022-07-28 17-42-56

## 6) Publisher Screen

Screenshot from 2022-07-28 14-54-00

Screenshot from 2022-07-28 14-48-40

## 7) ConfirmKeySend Screen

Screenshot from 2022-07-28 17-54-35

## 8) ConfirmPayment Screen

Screenshot from 2022-07-28 16-56-35

## 9) ConfirmSignMessage Screen

Screenshot from 2022-07-28 13-19-13

## 10) Send Satoshis (Home) Screen

<p>

</p>

### How has this been tested?

Tested manually

### Note
Have to check the success screens before merging!

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
My understanding was that we want to get rid of the background color because it can look very weird.

**Comment by im-adithya** [comment]
Made changes! Here's the new one:
Screenshot from 2022-07-07 16-48-36

File: src/app/components/NewPublisherCard/index.tsx
Comment (escapedcat) [inline src/app/components/NewPublisherCard/index.tsx:17]: Would be good to set an `alt` what's this supposed to display for semantic reasons.  
Instead of add the image via JS is it possible to default the image-prop with the default-image and overwrite it if a image-prop is being handed in?
Using JS to set an image seems unsemantic.
Maybe I don't understand the full context here.

File: src/app/components/NewPublisherCard/index.tsx
Comment (escapedcat) [inline src/app/components/NewPublisherCard/index.tsx:27]: Why do we need a title-attr on the h2 which is displaying the tile anyways?

**Review by escapedcat — CHANGES_REQUESTED** [review]

**Comment by bitcoinuser** [comment]
> ### Describe the changes you have made in this PR
> Redesign of "Enable" screen
> 
> ### Type of change (Remove other not matching type)
> * `chore`: Revamp (non-breaking change improving design)
> 
> ### Screenshots of the changes (If any)
> Screenshot from 2022-07-06 17-19-17
> 
> ### How has this been tested?
> Tested manually
> 
> ### Checklist
> * [x]  My code follows the style guidelines of this project and performed a self-review of my own code
> * [x]  New and existing tests pass locally with my changes
> * [x]  I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

Hi,

I was chatting with @bumi about my experience and doubts about this part of Alby and I would like to give some feedbacks.
Problems: 
1- I thought that the permission in case I have two accounts on Alby would be for the two accounts. Was not clear to me that is just for the account selected on top left.
2- It's easy for users forget to select the correct account on top left.

Feedback:

1- I think for this kind of window, it's not necessary show the account switch button that exist on top left. People usually ignore it when making actions on this kind of window.
2-I think should be able to grant permission for more than one account if more than one account exist on Alby.
3- Below the text (Request approval for transactions), should have a text like: (Select the accounts to grant permissions). The connect button should stay disabled until the user select the accounts that will receive grant permission.
4- Should be able to edit the grant permission if the user wants, like disable permission on a site for one or the two accounts. 

Thanks

**Comment by bitcoinuser** [comment]
> 3- Below the text (Request approval for transactions), should have a text like: (Select the accounts to grant permissions). The connect button should stay disabled until the user select the accounts that will receive grant permission.

Or if the user has two or more accounts, after clicking on connect button, Alby show a window warning the user has two accounts and give him the option to choose which ones he would like to grant permission.

**Comment by bitcoinuser** [comment]
Another idea is remove the Cancel button and rename the Connect button to Select accounts.

**Comment by escapedcat** [comment]
@bitcoinuser thanks for your feedback! We won't tackle this in the PR I believe. I'm worried your feedback might in this way.
Are you on the [bitcoin design slack](https://github.com/getAlby/lightning-browser-extension#join-the-conversation)? You could join our channel there and talk to @dvoroneca and bumi directly. They can open up an issue to tackle these things in the future if needed.

File: src/app/components/NewPublisherCard/index.tsx
Comment (im-adithya) [reply src/app/components/NewPublisherCard/index.tsx:17]: I have tried this in a project previously, but I think this kind of fallback using ternary would only work when the image URL is undefined; in cases where the link is valid but the image is broken (which can happen more often here), the onError works to replace the image with the fallback

File: src/app/components/NewPublisherCard/index.tsx
Comment (im-adithya) [reply src/app/components/NewPublisherCard/index.tsx:27]: Thought it would help on truncate, but ig the design allows more than 1 line of text! Changing it!

**Comment by escapedcat** [comment]
Tested with [starblocks.acinq.co](https://starblocks.acinq.co/):
- Should the URL have `test-overflow: ellipsis` and a title tag to show the full link there?
    image
- "Pay now" screen doesn't fit in the prompt. It's too high and cuts off some next below.
    The "old" prompt will show a scrollbar. This "new" one doesn't seem to do this.
    image image
    See old:
      image
    (Tested "old" on Chrome, "new" on Chromium. Not sure if that's an issue)
- Looks like all occurrences of ConformOrCancel/PublisherCard have been replaced, apart from this newly merged file:
    https://github.com/im-adithya/lightning-browser-extension/blob/task-revamp/src/app/screens/LNURLChannel.tsx  
    This can also use the new ones I assume.

File: src/app/screens/LNURLAuth.tsx
Comment (escapedcat) [inline src/app/screens/LNURLAuth.tsx:40]: What's google doing here? :)

**Review by escapedcat — CHANGES_REQUESTED** [review]

File: src/app/screens/LNURLAuth.tsx
Comment (im-adithya) [reply src/app/screens/LNURLAuth.tsx:40]: Oops! xD

**Comment by im-adithya** [comment]
Addressed all issues, ready for review!

**Comment by bumi** [comment]
good job @im-adithya 

@dvoroneca can you look at this, and use this? we need to see how we step by step implement the new components. 
to me the publisher screen takes too much screen space right now.

**Comment by escapedcat** [comment]
Actions to finish this PR:

1. @dvoroneca will have a look how to maybe adjust the banner
2. We need to come up with a final copy-text for:
     image

**Comment by dvoroneca** [comment]
### 1) Enable Screen

Reduce the card size, place it on a card
Connect with www podverce fm
----------
### 2) lnurlAuth Screen

Reduce the card size and add a layout variant
Send Payment

-----------
### 3) lnurlPay Screen
Reduce the card size and add a layout variant

Send Payment-3
------------

### 4) lnurlWithdraw Screen
Reduce the card size and add a layout variant

Send Payment-1

----------

### 5) MakeInvoice Screen

Reduce the card size and add a layout variant

Send Payment

-----------

### 6) Publisher Screen
# 👍

-----------

### 7) ConfirmKeySend Screen

Reduce the card size and add a layout variant

Send Payment

-------------

### 8) ConfirmPayment Screen

Reduce the card size and add a layout variant

Send Payment-1

---------------

### 9) ConfirmSignMessage Screen

Reduce the card size and add a layout variant

Send Payment-6

--------------

### 10) Send Satoshis (Home) Screen
# 👍

**Comment by dvoroneca** [comment]
> Actions to finish this PR:
> 
> 1. @dvoroneca will have a look how to maybe adjust the banner
> 2. We need to come up with a final copy-text for:
>    image
Connect with www podverce fm

**Comment by bumi** [comment]
@dvoroneca thanks, placing it on a card is also progress towards your next screens that show the sent amount and the recipient?

**Comment by im-adithya** [comment]
> ### 9) ConfirmSignMessage Screen
> Reduce the card size and add a layout variant
> 
> Send Payment-6

The Withdraw heading is an error, I assume?

**Comment by im-adithya** [comment]
Screenshot from 2022-08-02 16-33-11
Screenshot from 2022-08-02 16-37-50
<p>




</p>
<p>


</p>

Screenshot from 2022-08-02 16-34-31

Screenshot from 2022-08-02 16-34-40

**Comment by dvoroneca** [comment]
Nice, thanks. You've also edited the headers 👍

**Comment by MoritzKa** [comment]
While testing I experienced this behavior:

Chrome: Uploading chrome_OHbX04ad8U.mp4…

Firefox: jp8mbT7242


I used these [nightlies](https://github.com/getAlby/lightning-browser-extension/actions/runs/2781795078)

File: src/app/router/Prompt/Prompt.tsx
Comment (bumi) [inline src/app/router/Prompt/Prompt.tsx:60]: be very very very careful with such test data hacks. and actually never use those. if you need something like that then we need to look at the flow to make it easier that in development this actually can be tested. 

using the app in a real world context is most important also during development.

**Review by bumi — CHANGES_REQUESTED** [review]

File: src/app/components/ConfirmOrCancel/index.tsx
Comment (bumi) [inline src/app/components/ConfirmOrCancel/index.tsx:40]: I think the block section we only need on the enable screen.
thus we probably don't need that in here, do we?

Comment (im-adithya) [reply src/app/components/ConfirmOrCancel/index.tsx:40]: Yes, but the catch is that the link should be placed inside the component...
Can passing it as a child work?

**Comment by im-adithya** [comment]
> While testing I experienced this behavior:
> 
> Chrome:
> 
>  chrome_OHbX04ad8U.mp4 
> Firefox: jp8mbT7242
> 
> I used these [nightlies](https://github.com/getAlby/lightning-browser-extension/actions/runs/2781795078)

Sorry for that, it should work now!

File: src/app/components/ConfirmOrCancel/index.tsx
Comment (bumi) [reply src/app/components/ConfirmOrCancel/index.tsx:40]: what do you mean with inside the component? 

What I mean is that we do not have it inside the component, but just add it to the enable screen - or wherever it is needed

Comment (im-adithya) [reply src/app/components/ConfirmOrCancel/index.tsx:40]: But currently, we are using flex and justify-between to push the ConfirmOrCancel screen to the bottom, so we would have to have the block link inside it, otherwise, the workaround would be to use a prop to disable showing the usual "only_trusted" string and then add it separately and wrap the component and block link in a div. Should I do this, then?

Screenshot from 2022-08-02 16-33-11 (copy)

Comment (bumi) [reply src/app/components/ConfirmOrCancel/index.tsx:40]: no I would not add new props. Rather less props / less options and less code is always better (less complexity that needs to be managed) 

@escapedcat do you have an idea? 

I would actually like to have those text out of the ConfirmOrCancel component and also have the "block" option only on the enable screen. So I don't think the ConfirmOrCancel should know about the blocking

**Comment by reneaaron** [comment]
First of all: I think this is a huge improvement, especially the enable flows and publisher cards look a lot cleaner to me. 💯

I noticed some minor things: 

1) Default screen for publishers with tipping 
image

While the button seems to be a bit odd in this screen. Maybe we could also use the header from here? 
image

2) Rounding of logo container
I've seen different cases where the logo of the publisher is cropped quite badly: 
image
image

Do you think we could go with lesser rounding here so those logos do not get cropped anymore? (here with `rounded-lg`) /cc @dvoroneca 

image

**Comment by dvoroneca** [comment]
Thanks @reneaaron for pointing this issues out. Let's redefine the logo cropping to 8 px corner radius and see if this looks better. 

About the button, also agree that instances that have a button should have a wide state.

👍

**Comment by im-adithya** [comment]
> Default screen for publishers with tipping

Fixed! The actual idea was to have the second screen, but somehow it didn't fetch the description properly, resulting in the small single-row component.

> Let's redefine the logo cropping to 8 px corner radius and see if this looks better.

Done!

**Comment by im-adithya** [comment]
> Default screen for publishers with tipping

Fixed! The actual idea was to have the second screen, but somehow it didn't fetch the description properly, resulting in the small single-row component.

> Let's redefine the logo cropping to 8 px corner radius and see if this looks better.

Done!

**Comment by MoritzKa** [comment]
If the website does not offer an og:image the payment button looks off
image

**Comment by MoritzKa** [comment]
if I log in to a website. Can we switch the button from "Connect" to "Login"
chrome_EA731kITrT

**Comment by MoritzKa** [comment]
I suspect the "do not ask for this site again" button can have bad UX effects
image

I clicked this button on https://getalby.com/demo and also confirmed the connection. Then I deleted the connection in the extension again. Since then I there is no Alby prompt window again if I press the "Pay 25 Sats" on that website. But since I deleted the connection I cannot use that button anymore.

**Comment by bumi** [comment]
> I clicked this button on https://getalby.com/demo and also confirmed the connection. Then I deleted the connection in the extension again. Since then I there is no Alby prompt window again if I press the "Pay 25 Sats" on that website. But since I deleted the connection I cannot use that button anymore.

not sure if I understand this, because it is the point of the block feature that you are not prompted anymore. 
but this is also unrelated to this PR and related to the blocklist feature - did you use this in the current version?

**Comment by escapedcat** [comment]
> > I clicked this button on https://getalby.com/demo and also confirmed the connection. Then I deleted the connection in the extension again. Since then I there is no Alby prompt window again if I press the "Pay 25 Sats" on that website. But since I deleted the connection I cannot use that button anymore.

@MoritzKa will be tackled here: https://github.com/getAlby/lightning-browser-extension/issues/1225
I can do that right after we merge this.

**Comment by MoritzKa** [comment]
> > I clicked this button on https://getalby.com/demo and also confirmed the connection. Then I deleted the connection in the extension again. Since then I there is no Alby prompt window again if I press the "Pay 25 Sats" on that website. But since I deleted the connection I cannot use that button anymore.
> 
> not sure if I understand this, because it is the point of the block feature that you are not prompted anymore. but this is also unrelated to this PR and related to the blocklist feature - did you use this in the current version?

I thought I could not pay on the website anymore if I block further requests. But I did not see the "Enable now" button in the extension. All good!

File: src/app/screens/MakeInvoice.tsx
Comment (bumi) [inline src/app/screens/MakeInvoice.tsx:90]: should this be a header component? 
we have some duplication there in div and css classes.

File: src/app/components/Button/index.tsx
Comment (bumi) [inline src/app/components/Button/index.tsx:37]: hmmm, I think this is a bit hard to read
what means fullWidth vs. halfWidth
and why can both be true?

**Comment by MoritzKa** [comment]
Not super bad; just one observation I had. 
I enable Alby on a website (e.g. makers.bolt.fun); then I delete the connection in the extension window; but I can still use the website and make a payment. Only when I reload the page the "enable" prompt appears again. 
I would expect that the "enable" prompt appears with having to reload the page.

https://user-images.githubusercontent.com/13707997/182706173-0ebee45a-9e5c-43d0-ab96-7a218c87b11c.mp4

**Comment by bumi** [comment]
that has always been like that. and you're right that's not ideal. can you create a new issue (this is not part of this PR)

**Comment by bumi** [comment]
some suggestions @im-adithya @escapedcat: https://github.com/im-adithya/lightning-browser-extension/pull/6

File: src/app/components/PublisherCard/index.tsx
Comment (escapedcat) [inline src/app/components/PublisherCard/index.tsx:27]: See if you can use the existing `classNames` helper like i.e. here:  
https://github.com/getAlby/lightning-browser-extension/blob/feat/1216_refactor-prompt-popup-lnurl-handling/src/app/components/Button/index.tsx#L29

File: src/app/components/PublisherCard/index.tsx
Comment (escapedcat) [inline src/app/components/PublisherCard/index.tsx:35]: For these smaller optional classes you can use template literals like i.e. here:
https://github.com/getAlby/lightning-browser-extension/blob/feat/1216_refactor-prompt-popup-lnurl-handling/src/app/components/Badge/index.tsx#L11

File: src/app/screens/ConfirmKeysend/index.tsx
Comment (escapedcat) [inline src/app/screens/ConfirmKeysend/index.tsx:110]: As bumi mentioned below. Looks like the header-component HTML is the same for all screens. Just the text is changing, right? If that's the case we should move it into it's own component.

File: src/app/screens/ConfirmPayment/index.tsx
Comment (escapedcat) [inline src/app/screens/ConfirmPayment/index.tsx:108]: See above

File: src/app/screens/ConfirmSignMessage/index.tsx
Comment (escapedcat) [inline src/app/screens/ConfirmSignMessage/index.tsx:59]: See above

File: src/app/screens/ConfirmSignMessage/index.tsx
Comment (escapedcat) [inline src/app/screens/ConfirmSignMessage/index.tsx:68]: Kinda looks like we can also move this into it's own component to keep all views consistent.

**Review by escapedcat — COMMENTED** [review]
Moving out components doesn't need to block this PR for the release I think. I'm ok with merging this if everything else works and we tackle the component split in a follow-up PR.

File: src/app/components/Button/index.tsx
Comment (im-adithya) [reply src/app/components/Button/index.tsx:37]: I didn't understand the question, here we check if neither of the two is set and apply the original thing just so as not to break things.

Earlier (before this PR):
`fullWidth ? "px-0 py-2" : "px-7 py-2"`

Full Width Button
181014351-cab1358f-9f30-47e1-97af-39ec7547cc38

Half Width Button
Screenshot from 2022-08-04 12-42-47

**Comment by dvoroneca** [comment]
### Hi guys, happy this is merged now, however, there are a few tweaks regarding this update.

1. Too many symbols in a user profile name. Should be only one line of text.
Screenshot 2022-08-04 at 16 38 1

Group 27

-----------

2. The Header is still 20px text but should be 18px.

Screenshot 2022-08-04 at 17 28 33

**Pull request merged**
